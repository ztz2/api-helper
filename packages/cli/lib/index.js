"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.formatCode = exports.run = exports.defineConfig = exports.checkIsInterface = void 0;
const types_1 = require("../lib/types");
const Service_1 = __importDefault(require("../lib/service/Service"));
const format_code_1 = __importDefault(require("../lib/tools/format-code"));
exports.formatCode = format_code_1.default;
__exportStar(require("./helpers"), exports);
var util_1 = require("@api-helper/template/lib/utils/util");
Object.defineProperty(exports, "checkIsInterface", { enumerable: true, get: function () { return util_1.checkIsInterface; } });
function defineConfig(config) {
    return Array.isArray(config) ? config : [config];
}
exports.defineConfig = defineConfig;
function run(cmd = null, configFilePath, isTestEnv = false) {
    return __awaiter(this, void 0, void 0, function* () {
        switch (cmd) {
            case 'init':
                Service_1.default.init();
                break;
            default:
                yield new Service_1.default(configFilePath, isTestEnv).run();
        }
    });
}
exports.run = run;
