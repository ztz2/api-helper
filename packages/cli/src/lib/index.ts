import {
  Config,
  DocumentResourceList,
  DocumentParsedList,
} from './types';
import Service from '@/lib/service/Service';

export type { Config, DocumentResourceList, DocumentParsedList };

export function defineConfig(config: Config | Config[]): Config[] {
  return Array.isArray(config) ? config : [config];
}

export async function run(cmd: 'init' | 'root', configFilePath?: string) {
  switch (cmd) {
    case 'init':
      await Service.init();
      break;
    default:
      await new Service(configFilePath).run();
  }
}
