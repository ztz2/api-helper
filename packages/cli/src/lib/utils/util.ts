import { dirname, relative } from 'path';

/**
 * @description 将路径统一为 unix 风格的路径。
 * @param {string} path 路径
 * @return {string} unix 风格的路径
 */
export function toUnixPath(path: string) {
  return path.replace(/[/\\]+/g, '/')
}

/**
 * @description 获得规范化的相对路径。
 * @param {string} from 来源路径
 * @param {string} to 去向路径
 * @return { string } 相对路径
 */
export function getNormalizedRelativePath(from: string, to: string) {
  return toUnixPath(relative(dirname(from), to))
    .replace(/^(?=[^.])/, './')
    .replace(/\.(ts|js)x?$/i, '')
}

export function checkType(value: any, target: string) {
  return Object.prototype.toString.call(value) === `[object ${target}]`;
}
