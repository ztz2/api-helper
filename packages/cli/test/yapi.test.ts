import { readFile } from 'fs-extra';
import { join, resolve } from 'path';

import { run } from '@root/src/lib';
import { createTempFile } from '@root/src/lib/tools/util';

describe('Service', () => {

  test('生成 typescript api 代码', async () => {
    const configFile = createTempFile(`
import { resolve } from 'path';
export default {
  documentServers: [
    {
      url: 'https://yapi.pro',
      type: 'yapi',
      authToken: '2e7e8af1073feb9e629a743884529ddc7b6e67a1decaa52e28757f24739fab28',
      dataKey: ''
    }
  ],
  requestFunctionFilePath: '${join(process.cwd(), './test/__temp__/request.ts').replace(/\\/gim, '\\\\')}',
  output: {
    path: '${join(process.cwd(), './test/__temp__').replace(/\\/gim, '\\\\')}',
    filename: 'api.ts',
  }
};`);
    await run('root', configFile);
    const code = await readFile(resolve(__dirname, './__temp__/api.ts'));
    expect(code.toString()).toMatchSnapshot('生成 typescript api 代码');
  });
});
