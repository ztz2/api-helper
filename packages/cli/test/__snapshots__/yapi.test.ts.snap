// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`yapi文档 基于yapi，生成 typescript api 代码: 基于yapi，生成 typescript api 代码 1`] = `
"/* tslint:disable */
/* eslint-disable */
/* prettier-ignore-start */

/** ————————————— 警告：该文件由 api-helper 自动生成，切记不能对文件进行修改（因为生成之后内容将会覆盖该文件，如果自动生成代码不满足业务，提PR） —————————————  **/

// @ts-ignore
// prettier-ignore
import {
  RequestFunctionRestArgsType,
  processRequestFunctionConfig,
} from '@api-helper/core/es/lib/helpers';
// @ts-ignore
// prettier-ignore
import request from './request';
// @ts-ignore
// prettier-ignore
type CurrentRequestFunctionRestArgsType = RequestFunctionRestArgsType<typeof request>;

/**
 * @description uploads an image【请求数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969388
 * @url [ POST ] /v2/pet/{petId}/uploadImage
 */
export interface IV2PetPetIdUploadImageRequestByPost {
  // ID of pet to update
  petId: string;
  // Additional data to pass to server
  additionalMetadata?: string;
  // file to upload
  file?: string;
}
/**
 * @description uploads an image【响应数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969388
 * @url [ POST ] /v2/pet/{petId}/uploadImage
 */
export interface IV2PetPetIdUploadImageResponseByPost {
  code: number;
  type: string;
  message: string;
}
/**
 * @description uploads an image
 * @doc https://yapi.pro/project/16095/interface/api/969388
 * @url [ POST ] /v2/pet/{petId}/uploadImage
 */
export function v2PetPetIdUploadImageByPost(
  data: IV2PetPetIdUploadImageRequestByPost,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<IV2PetPetIdUploadImageResponseByPost>(
    processRequestFunctionConfig(
      data,
      extraData,
      v2PetPetIdUploadImageByPost.requestConfig
    ),
    ...args
  );
}
v2PetPetIdUploadImageByPost.requestConfig = {
  path: '/v2/pet/{petId}/uploadImage',
  method: 'POST',
  formDataKeyNameList: ['additionalMetadata', 'file'],
  pathParamKeyNameList: ['petId'],
  queryStringKeyNameList: [],
};

/**
 * @description Add a new pet to the store【请求数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969390
 * @url [ POST ] /v2/pet
 */
export interface IV2PetRequestByPost {
  id?: number;
  category: {
    id?: number;
    name?: string;
  };
  name: string;
  photoUrls: Array<string>;
  tags: Array<{
    id?: number;
    name?: string;
  }>;
  // pet status in the store
  status?: 'available' | 'pending' | 'sold';
}
/**
 * @description Add a new pet to the store【响应数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969390
 * @url [ POST ] /v2/pet
 */
export type V2PetResponseTypeByPost = any;
/**
 * @description Add a new pet to the store
 * @doc https://yapi.pro/project/16095/interface/api/969390
 * @url [ POST ] /v2/pet
 */
export function v2PetByPost(
  data: IV2PetRequestByPost,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<V2PetResponseTypeByPost>(
    processRequestFunctionConfig(data, extraData, v2PetByPost.requestConfig),
    ...args
  );
}
v2PetByPost.requestConfig = {
  path: '/v2/pet',
  method: 'POST',
  formDataKeyNameList: [],
  pathParamKeyNameList: [],
  queryStringKeyNameList: [],
};

/**
 * @description Update an existing pet【请求数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969392
 * @url [ PUT ] /v2/pet
 */
export interface IV2PetRequestByPut {
  id?: number;
  category: {
    id?: number;
    name?: string;
  };
  name: string;
  photoUrls: Array<string>;
  tags: Array<{
    id?: number;
    name?: string;
  }>;
  // pet status in the store
  status?: 'available' | 'pending' | 'sold';
}
/**
 * @description Update an existing pet【响应数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969392
 * @url [ PUT ] /v2/pet
 */
export type V2PetResponseTypeByPut = any;
/**
 * @description Update an existing pet
 * @doc https://yapi.pro/project/16095/interface/api/969392
 * @url [ PUT ] /v2/pet
 */
export function v2PetByPut(
  data: IV2PetRequestByPut,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<V2PetResponseTypeByPut>(
    processRequestFunctionConfig(data, extraData, v2PetByPut.requestConfig),
    ...args
  );
}
v2PetByPut.requestConfig = {
  path: '/v2/pet',
  method: 'PUT',
  formDataKeyNameList: [],
  pathParamKeyNameList: [],
  queryStringKeyNameList: [],
};

/**
 * @description Finds Pets by status【请求数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969394
 * @url [ GET ] /v2/pet/findByStatus
 */
export interface IV2PetFindByStatusRequestByGet {
  // Status values that need to be considered for filter
  status: string;
}
/**
 * @description Finds Pets by status【响应数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969394
 * @url [ GET ] /v2/pet/findByStatus
 */
export type V2PetFindByStatusResponseTypeByGet = Array<{
  id: number;
  category: {
    id: number;
    name: string;
  };
  name: string;
  photoUrls: Array<string>;
  tags: Array<{
    id: number;
    name: string;
  }>;
  // pet status in the store
  status: 'available' | 'pending' | 'sold';
}>;
/**
 * @description Finds Pets by status
 * @doc https://yapi.pro/project/16095/interface/api/969394
 * @url [ GET ] /v2/pet/findByStatus
 */
export function v2PetFindByStatusByGet(
  data: IV2PetFindByStatusRequestByGet,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<V2PetFindByStatusResponseTypeByGet>(
    processRequestFunctionConfig(
      data,
      extraData,
      v2PetFindByStatusByGet.requestConfig
    ),
    ...args
  );
}
v2PetFindByStatusByGet.requestConfig = {
  path: '/v2/pet/findByStatus',
  method: 'GET',
  formDataKeyNameList: [],
  pathParamKeyNameList: [],
  queryStringKeyNameList: ['status'],
};

/**
 * @description Finds Pets by tags【请求数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969396
 * @url [ GET ] /v2/pet/findByTags
 */
export interface IV2PetFindByTagsRequestByGet {
  // Tags to filter by
  tags: string;
}
/**
 * @description Finds Pets by tags【响应数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969396
 * @url [ GET ] /v2/pet/findByTags
 */
export type V2PetFindByTagsResponseTypeByGet = Array<{
  id: number;
  category: {
    id: number;
    name: string;
  };
  name: string;
  photoUrls: Array<string>;
  tags: Array<{
    id: number;
    name: string;
  }>;
  // pet status in the store
  status: 'available' | 'pending' | 'sold';
}>;
/**
 * @description Finds Pets by tags
 * @doc https://yapi.pro/project/16095/interface/api/969396
 * @url [ GET ] /v2/pet/findByTags
 */
export function v2PetFindByTagsByGet(
  data: IV2PetFindByTagsRequestByGet,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<V2PetFindByTagsResponseTypeByGet>(
    processRequestFunctionConfig(
      data,
      extraData,
      v2PetFindByTagsByGet.requestConfig
    ),
    ...args
  );
}
v2PetFindByTagsByGet.requestConfig = {
  path: '/v2/pet/findByTags',
  method: 'GET',
  formDataKeyNameList: [],
  pathParamKeyNameList: [],
  queryStringKeyNameList: ['tags'],
};

/**
 * @description Find pet by ID【请求数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969398
 * @url [ GET ] /v2/pet/{petId}
 */
export interface IV2PetPetIdRequestByGet {
  // ID of pet to return
  petId: string;
}
/**
 * @description Find pet by ID【响应数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969398
 * @url [ GET ] /v2/pet/{petId}
 */
export interface IV2PetPetIdResponseByGet {
  id: number;
  category: {
    id: number;
    name: string;
  };
  name: string;
  photoUrls: Array<string>;
  tags: Array<{
    id: number;
    name: string;
  }>;
  // pet status in the store
  status: 'available' | 'pending' | 'sold';
}
/**
 * @description Find pet by ID
 * @doc https://yapi.pro/project/16095/interface/api/969398
 * @url [ GET ] /v2/pet/{petId}
 */
export function v2PetPetIdByGet(
  data: IV2PetPetIdRequestByGet,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<IV2PetPetIdResponseByGet>(
    processRequestFunctionConfig(
      data,
      extraData,
      v2PetPetIdByGet.requestConfig
    ),
    ...args
  );
}
v2PetPetIdByGet.requestConfig = {
  path: '/v2/pet/{petId}',
  method: 'GET',
  formDataKeyNameList: [],
  pathParamKeyNameList: ['petId'],
  queryStringKeyNameList: [],
};

/**
 * @description Updates a pet in the store with form data【请求数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969400
 * @url [ POST ] /v2/pet/{petId}
 */
export interface IV2PetPetIdRequestByPost {
  // ID of pet that needs to be updated
  petId: string;
  // Updated name of the pet
  name?: string;
  // Updated status of the pet
  status?: string;
}
/**
 * @description Updates a pet in the store with form data【响应数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969400
 * @url [ POST ] /v2/pet/{petId}
 */
export type V2PetPetIdResponseTypeByPost = any;
/**
 * @description Updates a pet in the store with form data
 * @doc https://yapi.pro/project/16095/interface/api/969400
 * @url [ POST ] /v2/pet/{petId}
 */
export function v2PetPetIdByPost(
  data: IV2PetPetIdRequestByPost,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<V2PetPetIdResponseTypeByPost>(
    processRequestFunctionConfig(
      data,
      extraData,
      v2PetPetIdByPost.requestConfig
    ),
    ...args
  );
}
v2PetPetIdByPost.requestConfig = {
  path: '/v2/pet/{petId}',
  method: 'POST',
  formDataKeyNameList: ['name', 'status'],
  pathParamKeyNameList: ['petId'],
  queryStringKeyNameList: [],
};

/**
 * @description Deletes a pet【请求数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969402
 * @url [ DELETE ] /v2/pet/{petId}
 */
export interface IV2PetPetIdRequestByDelete {
  // Pet id to delete
  petId: string;
}
/**
 * @description Deletes a pet【响应数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969402
 * @url [ DELETE ] /v2/pet/{petId}
 */
export type V2PetPetIdResponseTypeByDelete = any;
/**
 * @description Deletes a pet
 * @doc https://yapi.pro/project/16095/interface/api/969402
 * @url [ DELETE ] /v2/pet/{petId}
 */
export function v2PetPetIdByDelete(
  data: IV2PetPetIdRequestByDelete,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<V2PetPetIdResponseTypeByDelete>(
    processRequestFunctionConfig(
      data,
      extraData,
      v2PetPetIdByDelete.requestConfig
    ),
    ...args
  );
}
v2PetPetIdByDelete.requestConfig = {
  path: '/v2/pet/{petId}',
  method: 'DELETE',
  formDataKeyNameList: [],
  pathParamKeyNameList: ['petId'],
  queryStringKeyNameList: [],
};

/**
 * @description Place an order for a pet【请求数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969404
 * @url [ POST ] /v2/store/order
 */
export interface IV2StoreOrderRequestByPost {
  id?: number;
  petId?: number;
  quantity?: number;
  shipDate?: string;
  // Order Status
  status?: 'placed' | 'approved' | 'delivered';
  // Order Tag
  tag: Array<
    // Order Tag Child
    Array<{
      tagName?: 'error' | 'waring' | 'info';
    }>
  >;
  complete?: boolean;
}
/**
 * @description Place an order for a pet【响应数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969404
 * @url [ POST ] /v2/store/order
 */
export interface IV2StoreOrderResponseByPost {
  id: number;
  petId: number;
  quantity: number;
  shipDate: string;
  // Order Status
  status: 'placed' | 'approved' | 'delivered';
  // Order Tag
  tag: Array<
    // Order Tag Child
    Array<{
      tagName: 'error' | 'waring' | 'info';
    }>
  >;
  complete: boolean;
}
/**
 * @description Place an order for a pet
 * @doc https://yapi.pro/project/16095/interface/api/969404
 * @url [ POST ] /v2/store/order
 */
export function v2StoreOrderByPost(
  data: IV2StoreOrderRequestByPost,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<IV2StoreOrderResponseByPost>(
    processRequestFunctionConfig(
      data,
      extraData,
      v2StoreOrderByPost.requestConfig
    ),
    ...args
  );
}
v2StoreOrderByPost.requestConfig = {
  path: '/v2/store/order',
  method: 'POST',
  formDataKeyNameList: [],
  pathParamKeyNameList: [],
  queryStringKeyNameList: [],
};

/**
 * @description Find purchase order by ID【请求数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969406
 * @url [ GET ] /v2/store/order/{orderId}
 */
export interface IV2StoreOrderOrderIdRequestByGet {
  // ID of pet that needs to be fetched
  orderId: string;
}
/**
 * @description Find purchase order by ID【响应数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969406
 * @url [ GET ] /v2/store/order/{orderId}
 */
export interface IV2StoreOrderOrderIdResponseByGet {
  id: number;
  petId: number;
  quantity: number;
  shipDate: string;
  // Order Status
  status: 'placed' | 'approved' | 'delivered';
  // Order Tag
  tag: Array<
    // Order Tag Child
    Array<{
      tagName: 'error' | 'waring' | 'info';
    }>
  >;
  complete: boolean;
}
/**
 * @description Find purchase order by ID
 * @doc https://yapi.pro/project/16095/interface/api/969406
 * @url [ GET ] /v2/store/order/{orderId}
 */
export function v2StoreOrderOrderIdByGet(
  data: IV2StoreOrderOrderIdRequestByGet,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<IV2StoreOrderOrderIdResponseByGet>(
    processRequestFunctionConfig(
      data,
      extraData,
      v2StoreOrderOrderIdByGet.requestConfig
    ),
    ...args
  );
}
v2StoreOrderOrderIdByGet.requestConfig = {
  path: '/v2/store/order/{orderId}',
  method: 'GET',
  formDataKeyNameList: [],
  pathParamKeyNameList: ['orderId'],
  queryStringKeyNameList: [],
};

/**
 * @description Delete purchase order by ID【请求数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969408
 * @url [ DELETE ] /v2/store/order/{orderId}
 */
export interface IV2StoreOrderOrderIdRequestByDelete {
  // ID of the order that needs to be deleted
  orderId: string;
}
/**
 * @description Delete purchase order by ID【响应数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969408
 * @url [ DELETE ] /v2/store/order/{orderId}
 */
export type V2StoreOrderOrderIdResponseTypeByDelete = any;
/**
 * @description Delete purchase order by ID
 * @doc https://yapi.pro/project/16095/interface/api/969408
 * @url [ DELETE ] /v2/store/order/{orderId}
 */
export function v2StoreOrderOrderIdByDelete(
  data: IV2StoreOrderOrderIdRequestByDelete,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<V2StoreOrderOrderIdResponseTypeByDelete>(
    processRequestFunctionConfig(
      data,
      extraData,
      v2StoreOrderOrderIdByDelete.requestConfig
    ),
    ...args
  );
}
v2StoreOrderOrderIdByDelete.requestConfig = {
  path: '/v2/store/order/{orderId}',
  method: 'DELETE',
  formDataKeyNameList: [],
  pathParamKeyNameList: ['orderId'],
  queryStringKeyNameList: [],
};

/**
 * @description Returns pet inventories by status【请求数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969410
 * @url [ GET ] /v2/store/inventory
 */
export type V2StoreInventoryRequestTypeByGet = any;
/**
 * @description Returns pet inventories by status【响应数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969410
 * @url [ GET ] /v2/store/inventory
 */
export type V2StoreInventoryResponseTypeByGet = any;
/**
 * @description Returns pet inventories by status
 * @doc https://yapi.pro/project/16095/interface/api/969410
 * @url [ GET ] /v2/store/inventory
 */
export function v2StoreInventoryByGet(
  data: V2StoreInventoryRequestTypeByGet,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<V2StoreInventoryResponseTypeByGet>(
    processRequestFunctionConfig(
      data,
      extraData,
      v2StoreInventoryByGet.requestConfig
    ),
    ...args
  );
}
v2StoreInventoryByGet.requestConfig = {
  path: '/v2/store/inventory',
  method: 'GET',
  formDataKeyNameList: [],
  pathParamKeyNameList: [],
  queryStringKeyNameList: [],
};

/**
 * @description Creates list of users with given input array【请求数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969412
 * @url [ POST ] /v2/user/createWithArray
 */
export interface IV2UserCreateWithArrayRequestByPost {
  // The name that needs to be fetched. Use user1 for testing.
  userId: string;
  // List of user object
  userName: string;
  // The password for login in clear text
  password: string;
}
/**
 * @description Creates list of users with given input array【响应数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969412
 * @url [ POST ] /v2/user/createWithArray
 */
export type V2UserCreateWithArrayResponseTypeByPost = any;
/**
 * @description Creates list of users with given input array
 * @doc https://yapi.pro/project/16095/interface/api/969412
 * @url [ POST ] /v2/user/createWithArray
 */
export function v2UserCreateWithArrayByPost(
  data: IV2UserCreateWithArrayRequestByPost,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<V2UserCreateWithArrayResponseTypeByPost>(
    processRequestFunctionConfig(
      data,
      extraData,
      v2UserCreateWithArrayByPost.requestConfig
    ),
    ...args
  );
}
v2UserCreateWithArrayByPost.requestConfig = {
  path: '/v2/user/createWithArray',
  method: 'POST',
  formDataKeyNameList: [],
  pathParamKeyNameList: ['userId'],
  queryStringKeyNameList: ['userName', 'password'],
};

/**
 * @description Creates list of users with given input array【请求数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969414
 * @url [ POST ] /v2/user/createWithList
 */
export type V2UserCreateWithListRequestTypeByPost = Array<{
  id?: number;
  username?: string;
  firstName?: string;
  lastName?: string;
  email?: string;
  password?: string;
  phone?: string;
  // User Status
  userStatus?: number;
}>;
/**
 * @description Creates list of users with given input array【响应数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969414
 * @url [ POST ] /v2/user/createWithList
 */
export type V2UserCreateWithListResponseTypeByPost = any;
/**
 * @description Creates list of users with given input array
 * @doc https://yapi.pro/project/16095/interface/api/969414
 * @url [ POST ] /v2/user/createWithList
 */
export function v2UserCreateWithListByPost(
  data: V2UserCreateWithListRequestTypeByPost,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<V2UserCreateWithListResponseTypeByPost>(
    processRequestFunctionConfig(
      data,
      extraData,
      v2UserCreateWithListByPost.requestConfig
    ),
    ...args
  );
}
v2UserCreateWithListByPost.requestConfig = {
  path: '/v2/user/createWithList',
  method: 'POST',
  formDataKeyNameList: [],
  pathParamKeyNameList: [],
  queryStringKeyNameList: [],
};

/**
 * @description Get user by user name【请求数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969416
 * @url [ GET ] /v2/user/{username}
 */
export interface IV2UserUsernameRequestByGet {
  // The name that needs to be fetched. Use user1 for testing.
  username1: string;
  username: string;
}
/**
 * @description Get user by user name【响应数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969416
 * @url [ GET ] /v2/user/{username}
 */
export interface IV2UserUsernameResponseByGet {
  id: number;
  username: string;
  firstName: string;
  lastName: string;
  email: string;
  password: string;
  phone: string;
  // User Status
  userStatus: number;
}
/**
 * @description Get user by user name
 * @doc https://yapi.pro/project/16095/interface/api/969416
 * @url [ GET ] /v2/user/{username}
 */
export function v2UserUsernameByGet(
  data: IV2UserUsernameRequestByGet,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<IV2UserUsernameResponseByGet>(
    processRequestFunctionConfig(
      data,
      extraData,
      v2UserUsernameByGet.requestConfig
    ),
    ...args
  );
}
v2UserUsernameByGet.requestConfig = {
  path: '/v2/user/{username}',
  method: 'GET',
  formDataKeyNameList: [],
  pathParamKeyNameList: ['username1', 'username'],
  queryStringKeyNameList: [],
};

/**
 * @description Updated user【请求数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969418
 * @url [ PUT ] /v2/user/{username}
 */
export interface IV2UserUsernameRequestByPut {
  // name that need to be updated
  username: string;
  id?: number;
  firstName?: string;
  lastName?: string;
  email?: string;
  password?: string;
  phone?: string;
  // User Status
  userStatus?: number;
}
/**
 * @description Updated user【响应数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969418
 * @url [ PUT ] /v2/user/{username}
 */
export type V2UserUsernameResponseTypeByPut = any;
/**
 * @description Updated user
 * @doc https://yapi.pro/project/16095/interface/api/969418
 * @url [ PUT ] /v2/user/{username}
 */
export function v2UserUsernameByPut(
  data: IV2UserUsernameRequestByPut,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<V2UserUsernameResponseTypeByPut>(
    processRequestFunctionConfig(
      data,
      extraData,
      v2UserUsernameByPut.requestConfig
    ),
    ...args
  );
}
v2UserUsernameByPut.requestConfig = {
  path: '/v2/user/{username}',
  method: 'PUT',
  formDataKeyNameList: [],
  pathParamKeyNameList: ['username'],
  queryStringKeyNameList: [],
};

/**
 * @description Delete user【请求数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969420
 * @url [ DELETE ] /v2/user/{username}
 */
export interface IV2UserUsernameRequestByDelete {
  // The name that needs to be deleted
  username1: string;
  username: string;
}
/**
 * @description Delete user【响应数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969420
 * @url [ DELETE ] /v2/user/{username}
 */
export type V2UserUsernameResponseTypeByDelete = any;
/**
 * @description Delete user
 * @doc https://yapi.pro/project/16095/interface/api/969420
 * @url [ DELETE ] /v2/user/{username}
 */
export function v2UserUsernameByDelete(
  data: IV2UserUsernameRequestByDelete,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<V2UserUsernameResponseTypeByDelete>(
    processRequestFunctionConfig(
      data,
      extraData,
      v2UserUsernameByDelete.requestConfig
    ),
    ...args
  );
}
v2UserUsernameByDelete.requestConfig = {
  path: '/v2/user/{username}',
  method: 'DELETE',
  formDataKeyNameList: [],
  pathParamKeyNameList: ['username1', 'username'],
  queryStringKeyNameList: [],
};

/**
 * @description Logs user into the system【请求数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969422
 * @url [ GET ] /v2/user/login
 */
export interface IV2UserLoginRequestByGet {
  // The user name for login
  username: string;
  // The password for login in clear text
  password: string;
}
/**
 * @description Logs user into the system【响应数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969422
 * @url [ GET ] /v2/user/login
 */
export type V2UserLoginResponseTypeByGet = string;
/**
 * @description Logs user into the system
 * @doc https://yapi.pro/project/16095/interface/api/969422
 * @url [ GET ] /v2/user/login
 */
export function v2UserLoginByGet(
  data: IV2UserLoginRequestByGet,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<V2UserLoginResponseTypeByGet>(
    processRequestFunctionConfig(
      data,
      extraData,
      v2UserLoginByGet.requestConfig
    ),
    ...args
  );
}
v2UserLoginByGet.requestConfig = {
  path: '/v2/user/login',
  method: 'GET',
  formDataKeyNameList: [],
  pathParamKeyNameList: [],
  queryStringKeyNameList: ['username', 'password'],
};

/**
 * @description Logs out current logged in user session【请求数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969424
 * @url [ GET ] /v2/user/logout
 */
export type V2UserLogoutRequestTypeByGet = any;
/**
 * @description Logs out current logged in user session【响应数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969424
 * @url [ GET ] /v2/user/logout
 */
export type V2UserLogoutResponseTypeByGet = any;
/**
 * @description Logs out current logged in user session
 * @doc https://yapi.pro/project/16095/interface/api/969424
 * @url [ GET ] /v2/user/logout
 */
export function v2UserLogoutByGet(
  data: V2UserLogoutRequestTypeByGet,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<V2UserLogoutResponseTypeByGet>(
    processRequestFunctionConfig(
      data,
      extraData,
      v2UserLogoutByGet.requestConfig
    ),
    ...args
  );
}
v2UserLogoutByGet.requestConfig = {
  path: '/v2/user/logout',
  method: 'GET',
  formDataKeyNameList: [],
  pathParamKeyNameList: [],
  queryStringKeyNameList: [],
};

/**
 * @description Create user【请求数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969426
 * @url [ POST ] /v2/user
 */
export interface IV2UserRequestByPost {
  id?: number;
  username?: string;
  firstName?: string;
  lastName?: string;
  email?: string;
  password?: string;
  phone?: string;
  // User Status
  userStatus?: number;
}
/**
 * @description Create user【响应数据类型定义】
 * @doc https://yapi.pro/project/16095/interface/api/969426
 * @url [ POST ] /v2/user
 */
export type V2UserResponseTypeByPost = any;
/**
 * @description Create user
 * @doc https://yapi.pro/project/16095/interface/api/969426
 * @url [ POST ] /v2/user
 */
export function v2UserByPost(
  data: IV2UserRequestByPost,
  extraData?: unknown,
  ...args: CurrentRequestFunctionRestArgsType
) {
  return request<V2UserResponseTypeByPost>(
    processRequestFunctionConfig(data, extraData, v2UserByPost.requestConfig),
    ...args
  );
}
v2UserByPost.requestConfig = {
  path: '/v2/user',
  method: 'POST',
  formDataKeyNameList: [],
  pathParamKeyNameList: [],
  queryStringKeyNameList: [],
};
"
`;
